.github/workflows/pages-release.yml

```diff
@@ -0,0 +1,109 @@
+name: Github Pages Release
+
+on:
+  workflow_call:
+    inputs:
+      versionPrefix:
+        required: false
+        type: string
+        description: "Optional prefix to add before version number"
+        default: "v"
+
+      sign:
+        required: false
+        type: boolean
+        description: "Sign tags"
+        default: true
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: Cache dependencies
+        id: cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ./node_modules
+            ./client/node_modules
+          key: modules-${{ hashFiles('package-lock.json') }}
+      - name: npm install
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          make clean-npm
+      - name: npm test
+        run: |
+          npm test
+      - name: npm build
+        run: npm run build
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: './build/'
+  Semantic_Release:
```

Comment:
Yeah I'm re-using the semantic release steps, but I removed the GPG signing because it doesn't apply here so I don't think I can directly call common-release.